python/lsst/afw/table/schemaMapper/schemaMapperContinued.py

```diff
@@ -85,3 +85,59 @@ def addMapping(self, input, output=None, doReplace=True):
             doReplace = output
             output = None
         return input._addMappingTo(self, output, doReplace)
+
+    def __eq__(self, other):
+        """SchemaMappers are equal if their respective input and output
+        schemas are identical, and they have the same mappings defined.
+
+        Note: It was simpler to implement equality in python than in C++.
+        """
+        iSchema = self.getInputSchema()
+        oSchema = self.getOutputSchema()
+        if (not (iSchema.compare(other.getInputSchema(), Schema.IDENTICAL) == Schema.IDENTICAL and
+                 oSchema.compare(other.getOutputSchema(), Schema.IDENTICAL) == Schema.IDENTICAL)):
+            return False
+
+        for item in iSchema:
+            if self.isMapped(item.key) and other.isMapped(item.key):
+                if (self.getMapping(item.key) == other.getMapping(item.key)):
+                    continue
+                else:
+                    return False
+            elif (not self.isMapped(item.key)) and (not other.isMapped(item.key)):
+                continue
+            else:
+                return False
+
+        return True
+
+    def __reduce__(self):
+        """To support pickle."""
+        mappings = {}
+        for item in self.getInputSchema():
+            mappings[item.key] = self.getMapping(item.key)
+        return (makeSchemaMapper, (self.getInputSchema(), self.getOutputSchema(), mappings))
+
+
+def makeSchemaMapper(input, output, mappings):
```

Comment:
Is this function more generally useful? If so, consider listing it in `__all__`.

Reply:
Probably not: you have to get the mappings in order first. Because of that, I don't think it would save any code, since you'd have to make the mappings dict.